js/core/src/tracing.ts

```diff
@@ -35,18 +35,26 @@ let nodeOtelConfig: TelemetryConfig | null = null;
 
 const instrumentationKey = '__GENKIT_TELEMETRY_INSTRUMENTED';
 
-export function ensureBasicTelemetryInstrumentation() {
+export async function ensureBasicTelemetryInstrumentation() {
   if (global[instrumentationKey]) {
     return;
   }
-  enableTelemetry({});
-  global[instrumentationKey] = true;
+  await enableTelemetry({});
 }
 
 /**
  * Enables tracing and metrics open telemetry configuration.
  */
-export function enableTelemetry(telemetryConfig: TelemetryConfig) {
+export async function enableTelemetry(
+  telemetryConfig: TelemetryConfig | Promise<TelemetryConfig>
+) {
+  global[instrumentationKey] = true;
```

Comment:
Could there still be a race condition if telemetryConfig takes a long time to resolve from here?

Reply:
what might happen is that some telemetry might not get recorded until it competes initialization... but we can actually block requests on telemetry initialization... let me do that.